pkg/operator/ceph/file/controller.go

```diff
@@ -288,6 +288,17 @@ func (r *ReconcileCephFilesystem) reconcileCreateFilesystem(cephFilesystem *ceph
 		return reconcile.Result{}, errors.Wrapf(err, "failed to create filesystem %q", cephFilesystem.Name)
 	}
 
+	// Enable mirroring if needed
+	if r.clusterInfo.CephVersion.IsAtLeastPacific() {
```

Comment:
Shouldn't this be done by the new cephfs mirroring controller? I must be missing why we need the `mirroring.enabled` setting in the filesystem since we have the separate CRD.

Reply:
The CRD deploys the cephfs-mirror daemon, the CephFilesystem setting, turns on mirroring on that filesystem. So let's imagine we have multiple CephFilesystem deployed, we don't want all of them to be mirrored, that's why I'm basing the decision out of the CephFilesystem CRD and the CephFilesystemMirror CRD, which goal is to *only* deploy the cephfs-mirror at the moment. Later it will add/remove peers once available in Ceph.

Reply:
I don't see anything specific to this filesystem being enabled, I only see a merge module being enabled. Is there another setting coming that will be specific to the filesystem?

Reply:
Since enabling the module only needs to happen once for the entire cluster, this still seems like it belongs with the singleton mirroring CR instead of being enabled as a property on the filesystem. Why not enable the module with the mirroring CR, and get rid of the filesystem CR setting? With rbd mirroring there is a pool-specific setting so it makes sense to have the boolean on the pool CR. But if there is no filesystem-specific setting for mirroring, I don't see why a bool is needed on the filesystem CR.

Reply:
You don't see anything specific to the fs because I'm waiting for ceph/ceph#39050 to merge so I can turn on mirroring for the fs. Now, we just enable the mirroring module. Anyway, we will always need the fs setting to turn on mirroring.

Honestly, I have no preference, I can move this `cephclient.MgrEnableModule` to the CephFilesysteMirror CRD. Either way, we will have to enable it.

What do you think?

Hope that clarifies.

Reply:
Got it, this is basically a placeholder for the per-filesystem setting that will come later.
Whether to enable the merge module sounds fine to keep here, as discussed it could go either here or with the mirroring CR.